### ISSUE TITLE
random.randrange and random.randint mishandle large ints

### ISSUE THREAD

**Issue by ProfessorO** [skulpt/skulpt#1211]
When a number larger than 2^31 - 1 is passed as the upper limit to random.randrange or random.randint, (with 0 as the lower limit, both implicitly and explicitly), those functions sometimes return a negative number. That's too obvious a "limit" to be coincidental. I've gotten it to consistently fail (more than .1% of the time) by adding just 10,000 to 2^31, but I bet it always breaks past that point and I just haven't let my tests run long enough.

**Comment by s-cork** [comment]
From what I can tell the code base for random has been written with numbers less than Number.MAX_SAFE_INTEGER in mind. So this would likely require an update to that module to handle large integers.

**Comment by Huge** [comment]
I've been fighting a relevant issue with huge numbers, where the upper bound is `2**127 - 1`, which I'm able to `print` without a problem.
I asume my code fails in https://github.com/skulpt/skulpt/blob/b763d3e24860a8e3d303878e0035d8bad9f55578/src/lib/random.js#L268 .
I have only subtle awareness of Javascript, but I'd like to help in making this improvement/PR..is it the right way to just wrap all the calculations there into `BigInt()` or do we need to assemble the larger numbers from the 32b integers?

**Comment by s-cork** [comment]
`BigInt` is not something we recommend in the codebase as it's not backwards compatible for old browser. (I think it was only added to safari in the last couple of months)

There are examples in `src/lib/maths.js` where large numbers can passed to `factorial` or `gcd` and are handled by the `big_integer` interface.
https://github.com/skulpt/skulpt/blob/master/src/lib/math.js#L226

I'm also hoping that we can use the [`JSBI`](https://github.com/GoogleChromeLabs/jsbi#how) library at some point which has a nice functional approach to handling `BigInt` 

here's the example of math.js function with a version of the JSBI library
https://github.com/s-cork/skulpt/blob/prototypical_getsets_mappingproxy/src/lib/math.js#L226

**Comment by Huge** [comment]
> native `BigInt` is not something we recommend in the codebase as it's not backwards compatible for older browsers. (I think it was only added to safari in the last couple of months)

 True! ( https://en.wikipedia.org/wiki/Safari_version_history#Safari_14 )

I'll look into the `Sk.builtin.biginteger` way to solve it. Let's see how far I can get..:)
